src/router/routes/admin/admin-location.ts

```diff
@@ -56,20 +58,24 @@ export class AdminLocationController extends ParentRouter implements IExpressCon
    * @apiGroup Admin Location
    * @apiPermission DirectorPermission
    *
-   * // TODO: Update when implemented
-   * @apiDescription NOTE: This route is not implemented yet
+   * @apiParam {Number} limit=Math.inf Limit to a certain number of responses
+   * @apiParam {Number} offset=0 The offset to start retrieving users from. Useful for pagination
+   *
    * @apiUse AuthArgumentRequired
    * @apiSuccess {Location[]} Array of locations
    * @apiUse ResponseBodyDescription
    */
-  private getAllLocationsHandler(res: Response, next: NextFunction) {
-    // Location.getAll(req.uow, {
-    //   count: res.locals.limit,
-    //   startAt: res.locals.offset,
-    // })
-    //   .then(stream => streamHandler(stream, res, next))
-    //   .catch(err => errorHandler500(err, next));
-    next(new HttpError('This is not implemented yet', 501));
+  private async getAllLocationsHandler(res: Response, next: NextFunction) {
+    try {
+      const result = await this.locationDataMapper.getAll({
+        count: res.locals.limit,
```

Comment:
Can you ensure that wherever res.locals.limit is being used, it is actually populated? I believe that is done in the parent admin router, but I just want to be sure.

Reply:
Limit is already handled before being passed to the admin-location.ts by admin.ts, the parent router. Below is a snippet of the code that handles it.

Admin.ts - line 24-29
```suggestion
    if (!request.query.limit || parseInt(request.query.limit, 10)) {
      response.locals.limit = parseInt(request.query.limit, 10);
    } else {
      const error = new HttpError('Limit must be an integer', 400);
      return Util.standardErrorHandler(error, next);
    }
```

Reply:
It is already being populated and handled by the parent router, admin.ts on lines 25-29